### PR TITLE
Fix sonar code smells

### PR DESCRIPTION
**Pull request by Rapster**

### CODE REVIEW

**Review by melloware — APPROVED** [review]

File: primefaces/src/main/java/org/primefaces/component/fileupload/FileUploadBase.java [hunk 1]
```diff
@@ -491,12 +491,7 @@ public void setDropZone(String dropZone) {
     }
 
     public boolean isDisplayFilename() {
-        return ComponentUtils.eval(getStateHelper(), PropertyKeys.displayFilename, () -> {
-            if (getMode().equals("simple") && isAuto()) {
-                return false;
-            }
-            return true;
-        });
+        return ComponentUtils.eval(getStateHelper(), PropertyKeys.displayFilename, () -> !getMode().equals("simple") || !isAuto());
```
Comment (tandraschko) [inline primefaces/src/main/java/org/primefaces/component/fileupload/FileUploadBase.java:494]: IMO its not nice to read
`!(getMode().equals("simple") && isAuto())` would be ok for me if you would like to switch to 1 liner

Comment (tandraschko) [reply primefaces/src/main/java/org/primefaces/component/fileupload/FileUploadBase.java:494]: @Rapster it even breaks the functionality. Per default it should not be displayed when simple&&auto, not just one of both

Comment (Rapster) [reply primefaces/src/main/java/org/primefaces/component/fileupload/FileUploadBase.java:494]: Weird, the condition looks the same to me, and the MR passed before I merged. Let's see with the new MR

Comment (tandraschko) [reply primefaces/src/main/java/org/primefaces/component/fileupload/FileUploadBase.java:494]: could also be that i have a "thinking error" in my brain :D but with your new PR it just reads more cleaner

and we dont have a test for this functionality :/

**Pull request merged**